ApixMiddleware.php

```diff
@@ -39,18 +42,27 @@ public function __construct(
     }
 
     /**
-     * @param \Symfony\Component\HttpFoundation\Request $request
-     * @return \Psr\Http\Message\ResponseInterface
+     *
+     * @param \Symfony\Component\HttpKernel\Event\RequestEvent $event
      */
-    public function before(Request $request)
+    public function before(RequestEvent $event)
     {
+
+        $request = $event->getRequest();
+
+        // Short circuit if this is an OPTIONS or HEAD request.
+        if (in_array(
+            strtoupper($request->getMethod()),
+            ['OPTIONS', 'HEAD']
+        )) {
+            return;
+        }
+
         // Short circuit if there's no Apix-Ldp-Resource header.
         if (!$request->headers->has("Apix-Ldp-Resource")) {
-            $this->log->debug("Malformed request, no Apix-Ldp-Resource header present");
-            return new Response(
-                "Malformed request, no Apix-Ldp-Resource header present",
-                400
-            );
+            $this->log->debug("No Apix-Ldp-Resource header present, no fedora_resource set");
+            $request->attributes->set('fedora_resource', false);
+            return;
```

Comment:
In [poking around and running tests](https://github.com/Islandora/Crayfish-Commons/compare/3.x...adam-vessey:Crayfish-Commons:feature/more-streaming), I found that the we broke the tests here, where it tests for the 400 error being emitted... easy enough fix just setting the response with something like:

```diff
diff --git a/ApixMiddleware.php b/ApixMiddleware.php
index b75df0a..5a29209 100644
--- a/ApixMiddleware.php
+++ b/ApixMiddleware.php
@@ -62,6 +62,10 @@ class ApixMiddleware implements EventSubscriberInterface
         if (!$request->headers->has("Apix-Ldp-Resource")) {
             $this->log->debug("No Apix-Ldp-Resource header present, no fedora_resource set");
             $request->attributes->set('fedora_resource', false);
+           $event->setResponse(new Response(
+                "Malformed request, no Apix-Ldp-Resource header present",
+                400
+            ));
             return;
         }

```

Reply:
Looking to address in https://github.com/Islandora/Crayfish-Commons/pull/57